File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/permalink/pathTypes/ArchiveIndexPathType.java
package com.eden.orchid.posts.permalink.pathTypes;

import com.eden.orchid.api.theme.pages.OrchidPage;
import com.eden.orchid.posts.pages.PostArchivePage;
import com.eden.orchid.posts.permalink.PermalinkPathType;

import javax.inject.Inject;

public class ArchiveIndexPathType extends PermalinkPathType {

    @Inject
    public ArchiveIndexPathType() {
        super(100);
    }

    @Override
    public boolean acceptsKey(OrchidPage page, String key) {
        return key.equals("archiveIndex");
    }

    @Override
    public String format(OrchidPage page, String key) {
        if(page instanceof PostArchivePage) {

            int index = ((PostArchivePage) page).getIndex();

            return (index == 1) ? "" : "" + index;
        }

        return null;
    }

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/permalink/pathTypes/ArchiveIndexPathType.kt
package com.eden.orchid.posts.permalink.pathTypes

import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.posts.pages.PostArchivePage
import com.eden.orchid.posts.permalink.PermalinkPathType

import javax.inject.Inject

class ArchiveIndexPathType @Inject
constructor() : PermalinkPathType(100) {

    override fun acceptsKey(page: OrchidPage, key: String): Boolean {
        return key == "archiveIndex"
    }

    override fun format(page: OrchidPage, key: String): String? {
        if (page is PostArchivePage) {

            val index = page.index

            return if (index == 1) "" else "" + index
        }

        return null
    }

}

